### PR TITLE
RK-617 front end implementation for £0 Validation Page

### PR DESCRIPTION
**Pull request by shiromi**

### CODE REVIEW

File: app/controllers/ZeroValidationController.scala [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  formProvider: ZeroValidationFormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  view: ZeroValidationView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController with I18nSupport {
+
+  val form = formProvider()
+
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
+    implicit request =>
```
Comment (shinealex33) [inline app/controllers/ZeroValidationController.scala:49]: Please add server side events

File: app/controllers/ZeroValidationController.scala [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        case Some(value) => form.fill(value)
+      }
+
+      Ok(view(preparedForm, mode))
+  }
+
+  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      form.bindFromRequest().fold(
+        formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
```
Comment (shinealex33) [inline app/controllers/ZeroValidationController.scala:61]: same - events for form error

File: app/viewmodels/checkAnswers/ZeroValidationSummary.scala [hunk 3]
```diff
@@ -0,0 +1,44 @@
+/*
```
Comment (shinealex33) [inline app/viewmodels/checkAnswers/ZeroValidationSummary.scala:1]: Do we need this page ? Where do we use this ?

File: app/views/ZeroValidationView.scala.html [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    @formHelper(action = routes.ZeroValidationController.onSubmit(mode), 'autoComplete -> "off") {
+
+        @if(form.errors.nonEmpty) {
+            @govukErrorSummary(ErrorSummaryViewModel(form))
+        }
+
+        @govukRadios(
+            RadiosViewModel.yesNo(
+                field = form("value"),
+                legend = LegendViewModel(messages("zeroValidation.heading")).asPageHeading()
```
Comment (shinealex33) [inline app/views/ZeroValidationView.scala.html:42]: This should P right not hint as per Ben

**Review by shinealex33 — CHANGES_REQUESTED** [review]
Please look into comments

File: app/views/ZeroValidationView.scala.html (hunk 4 above)
Comment (shiromi) [reply app/views/ZeroValidationView.scala.html:42]: hint text  added in <p>
@contentHint = {
    <p class="govuk-hint">@messages("zeroValidation.hint")</p>
}

**Review by shinealex33 — APPROVED** [review]
LGTM

**Pull request merged**